rust/gitxetcore/src/command/login.rs

```diff
@@ -82,10 +87,167 @@ fn apply_config(
                 axe_config.axe_code = Some(axe_key);
             }
         }
+        if let Some(aws_access_key) = auth_check.aws_access_key {
+            if !aws_access_key.is_empty() {
+                user.aws_access_key = Some(aws_access_key);
+            }
+        }
+        if let Some(aws_secret_key) = auth_check.aws_secret_key {
+            if !aws_secret_key.is_empty() {
+                user.aws_secret_key = Some(aws_secret_key);
+            }
+        }
     }
     Ok(())
 }
 
+// Run AWS CLI to write an access key and secret key
+// Returns the profilename we wrote to
+fn write_aws_config(cfg: &Cfg, host: &str) -> errors::Result<String> {
+    if cfg.user.is_none() {
+        return Err(errors::GitXetRepoError::Other(
+            "No user configuration".to_string(),
+        ));
+    }
+    let user = &cfg.user.as_ref().unwrap();
+    let aws_access_key = user
+        .aws_access_key
+        .as_ref()
+        .ok_or(errors::GitXetRepoError::AuthError(anyhow!(
+            "Not authenticated. Run git-xet login first"
+        )))?;
+    let aws_secret_key = user
+        .aws_secret_key
+        .as_ref()
+        .ok_or(errors::GitXetRepoError::AuthError(anyhow!(
+            "Not authenticated. Run git-xet login first"
+        )))?;
+    // dots and dashes are ok
+    // this makes the profile name slightly more readable
+    let mut profilename = host.to_string();
+    profilename.retain(|x| x.is_alphanumeric() || x == '.' || x == '-');
```

Comment:
_ ?

Reply:
I don't actually know! I just manually tested . and -